### ISSUE TITLE
Pagination

### ISSUE THREAD

**Issue by ssbittle** [prenda-school/prenda-spark#60]
### Variations:
**Types**: `Left-Arrow`, `Center`, `More`, `Right-Arrow`
**Sizes**: NA

**Design reference:**
[Figma file - Pagination](https://www.figma.com/file/y6Ca8BdJtQEd3Of1saRieF/Spark-Design-System-Web?node-id=949%3A463)

**Comment by WilliamKelley** [comment]
### Feedback

@ssbittle 

- In the States block, missing 'focus' state (keyboard nav)

**Comment by WilliamKelley** [comment]
This isn't fleshed out at all, but I needed this one-off in Code Club and gave a very quick, rough first attempt -- can serve as a starting place for further discussion / initial implementation.

Material-UI: [Pagination (rounded)](https://material-ui.com/components/pagination/#rounded-pagination), PaginationItem

```jsx
// FILE: ~/src/PaginationItem.jsx

import { PaginationItem as MuiPaginationItem } from '@material-ui/lab';

const PaginationItem = styled(MuiPaginationItem)`
  &.MuiPaginationItem-outlined {
    background-color: white;
    border: solid 2px ${theme.palette.neutral.mediumGrey};
    border-radius: 0;
    color: ${theme.palette.background.lightContrastText};
    font-weight: 700px;
    font-size: 1rem;
    line-height: 1.25rem;

    &:hover {
      background-color: ${theme.palette.neutral.lightGrey};
      border: solid 2px ${theme.palette.neutral.mediumGrey};
    }
    &:focus {
      background-color: ${theme.palette.neutral.lightGrey};
      border: solid 2px ${theme.palette.tertiary.blue[4]};
    }

    &.Mui-selected {
      color: white;
      background-color: ${theme.palette.tertiary.blue[3]};
      border: solid 2px ${theme.palette.tertiary.blue[5]};
    }
  }

  &.MuiPaginationItem-sizeLarge {
    height: 3rem;
    min-width: 3rem;
  }
`;

const SparkPaginationItem = (props) => {
  return  (
    <PaginationItem
       disableRipple
       disableTouchRipple
       focusRipple={false}
       {...props}
    />
  );
};

export { SparkPaginationItem as PaginationItem };
```

```jsx
// FILE: ~/src/Pagination.jsx
import { Pagination as MuiPagination } from '@material-ui/lab';
import { PaginationItem } from './PaginationItem';

const SparkPagination = (props) => {
  return (
    <Pagination
      count={pageMax}
      variant='outlined'
      shape='rounded'
      size='large'
      renderItem={(item) => (
        <PaginationItem {...item} />
      )}
      {...props}
    />
  );
};

export { SparkPagination as Pagination };
```
```jsx
// FILE: story

...
const [page, setPage] = React.useState(1);
const handlePageChange = (event, newPage) => setPage(newPage);

return (
  <Pagination
    page={page}
    count={pageMax}
    onChange={handlePageChange}
  />
);
...
```
